mpmp/utilities/plot_utilities.py

```diff
@@ -0,0 +1,281 @@
+                                                verbose=True)
+        compare_results_df.rename(columns={'identifier': 'gene'}, inplace=True)
+        compare_results_df['nlog10_p'] = -np.log10(compare_results_df.corr_pval)
+
+        sns.scatterplot(data=compare_results_df, x='delta_mean', y='nlog10_p',
+                        hue='reject_null', hue_order=[False, True], ax=ax,
+                        legend=(ix == 0))
+
+        # add vertical line at 0
+        ax.axvline(x=0, linestyle='--', linewidth=1.25, color='black')
+
+        # add horizontal line at statistical significance threshold
+        l = ax.axhline(y=-np.log10(sig_alpha), linestyle='--', linewidth=1.25)
+
+        # label horizontal line with significance threshold
+        # (matplotlib makes this fairly difficult, sadly)
+        ax.text(0.5, -np.log10(sig_alpha)+0.01,
+                r'$\mathbf{{\alpha = {}}}$'.format(sig_alpha),
+                va='center', ha='center', color=l.get_color(),
+                backgroundcolor=ax.get_facecolor())
+
+        # label axes and set axis limits
+        ax.set_xlabel('AUPR({}) - AUPR(expression)'.format(training_data), size=14)
+        ax.set_ylabel(r'$-\log_{10}($adjusted $p$-value$)$', size=14)
+        ax.set_xlim(xlim)
+        ax.set_ylim(ylim)
+
+        # only add a legend to the first subplot
+        if ix == 0:
+            ax.legend(title=r'Reject $H_0$', loc='upper left',
+                      fontsize=14, title_fontsize=14)
+
+        ax.set_title(
+            r'Mutation prediction, expression vs. {}'.format(training_data),
+            size=14
+        )
+
+        # label genes and adjust text to not overlap
+        # automatic alignment isn't perfect, can align by hand if necessary
+        text_labels = _label_points_compare(
+                          compare_results_df['delta_mean'],
+                          compare_results_df['nlog10_p'],
+                          compare_results_df.gene,
+                          ax,
+                          sig_alpha)
+        adjust_text(text_labels,
+                    ax=ax,
+                    expand_text=(1., 1.),
+                    lim=5)
+
+        if verbose:
+            print('{}: {}/{}'.format(training_data,
+                                     np.count_nonzero(compare_results_df.reject_null),
+                                     compare_results_df.shape[0]))
+
+    return axarr
+
+
+def plot_boxes(results_df,
+               axarr,
+               training_data_map,
+               orientation='h',
+               verbose=False):
+    """Make a box plot comparing classifier results between data types.
+
+    Arguments
+    ---------
+    results_df (pd.DataFrame): dataframe with processed results/p-values
+    axarr (array of Axes): axes array from plt.subplots
+    training_data_map (dict): maps internal data type names to display names
+
+    Returns
+    -------
+    axarr (array of Axes): axes with plots
```

Comment:
I'm a little lost here. So is this returning a multi-panel figure? In this case it looks like there are 2 panels..?

Reply:
Yeah, exactly, this returns an array of the matplotlib axes in the multi-panel figure.

I actually don't think this function needs to return anything, because Python objects (like matplotlib `Array` or `Figure` objects) are mutable. This means if we make changes to them inside the function (like adding plots to the axes), the changes will persist outside of the function.

This is as opposed to something like a Python string, which is immutable so you'd have to reassign it like I'm doing here.